sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -758,8 +760,12 @@ abstract class CastSuiteBase extends SparkFunSuite with ExpressionEvalHelper {
       withSQLConf(SQLConf.LEGACY_COMPLEX_TYPES_TO_STRING.key -> legacyCast.toString) {
         val ret1 = cast(Literal.create((1, "a", 0.1)), StringType)
         checkEvaluation(ret1, s"${lb}1, a, 0.1$rb")
-        val ret2 = cast(Literal.create(Tuple3[Int, String, String](1, null, "a")), StringType)
-        checkEvaluation(ret2, s"${lb}1,${if (legacyCast) "" else " null"}, a$rb")
+        val ret2 = cast(
+          Literal.create(Tuple4[String, Int, String, String](null, 1, null, "a")),
```

Comment:
ditto.

Reply:
I have reverted the changes to the existing unit tests and added new ones.